.github/workflows/format.yml

```diff
@@ -0,0 +1,25 @@
+name: Format
+on: 
+  push:
+    branches:
+      - !main
+
+jobs:
+  analyze:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2     # foundry docs say v3 but npm install uses v2
+        with:
+          submodules: recursive       # ensure forge-std installed
+      - uses: bahmutov/npm-install@v1
+        with:
+          useLockFile: true
+        env:
+          NODE_ENV: development       # install all test tooling 
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      - name: format
+        run: forge fmt contracts
```

Comment:
Does workflow add a commit with formatted files  after its done processing?

Reply:
I cant actually get it to run on the files as a workflow. in CLI `forge fmt` works fine, but as part of the workflow it does not